modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -500,6 +501,29 @@ def validate_service_periods!(service_information)
         end
       end
 
+      def validate_form_526_location_codes!(service_information)
+        # only retrieve separation locations if we'll need them
+        need_locations = service_information['servicePeriods'].detect do |service_period|
+          Date.strptime(service_period['activeDutyEndDate'], '%m-%d-%Y') > Time.zone.today
+        end
```

Comment:
should we just not return if this is nil? we're checking `EndDate >` here and below we next if `EndDate <=` ?